code/game/objects/structures/grille.dm

```diff
@@ -52,6 +52,17 @@
 	. = ..()
 	update_icon()
 
+/obj/structure/grille/update_icon()
+	if(QDELETED(src) || broken)
+		return
+
+	var/ratio = obj_integrity / max_integrity
+	ratio = CEILING(ratio*4, 1) * 25
+
+	if(ratio > 50)
+		return
```

Comment:
is the `CEILING` here really necessary? surely you could remove that line all together and just do `if(ratio > 0.5) => return`

Reply:
It's not necessary and I have no idea why TG was using it since my understanding of CEILING is that it's used for math calculations(?), probably once again has something to do with the TG copy-pasting the window damage state code. The code now makes a heck of a lot more sense to me since I suck with math.